lib/client-base.js

```diff
@@ -287,7 +297,19 @@ class ClientBase extends EventEmitter {
 
 class Parser extends HTTPParser {
   constructor (client, socket) {
-    super(HTTPParser.RESPONSE)
+    /* istanbul ignore next */
+    if (NODE_MAJOR_VERSION >= 12) {
+      super()
+      this.initialize(
+        HTTPParser.RESPONSE,
+        {},
+        client[kMaxHeadersSize],
+        false,
+        0
+      )
```

Comment:
In node 12 you can still use the old parser using a command line flag.

Reply:
I think that was in order to avoid breaking. I don't think that's relevant here. Also I'm not sure if/how a library can inspect command line options like an internal module.

Reply:
I know, but this will fail if that option is specified :/.

Reply:
> I know, but this will fail if that option is specified :/.

Oh, any ideas? try/catch with a retry with option enabled?

Reply:
Can a check be implemented using:

https://nodejs.org/dist/latest-v12.x/docs/api/process.html#process_process_execargv

?